genesis/public/core/reference_test.go

```diff
@@ -81,3 +83,146 @@ func TestNewReferenceDomain_WithNoParent(t *testing.T) {
 
 	assert.Equal(t, expected, refDomain)
 }
+
+func TestReferenceDomain_GetClassID(t *testing.T) {
+	refDomain := newReferenceDomain(nil)
+	domainID := refDomain.GetClassID()
+	assert.Equal(t, class.ReferenceDomainID, domainID)
+}
+
+func TestReferenceDomain_RegisterReference(t *testing.T) {
+	domain := "1"
+	record := "1"
+	refObject, err := object.NewReference(record, domain, object.GlobalScope)
+	assert.NoError(t, err)
+
+	refDomain := newReferenceDomain(nil)
+	registered, err := refDomain.RegisterReference(refObject)
+
+	assert.NoError(t, err)
+
+	_, err = uuid.FromString(registered)
+	assert.NoError(t, err)
+}
+
+func TestReferenceDomain_ResolveReference(t *testing.T) {
+	domain := "1"
+	record := "1"
+	refObject, err := object.NewReference(record, domain, object.GlobalScope)
+	assert.NoError(t, err)
+
+	refDomain := newReferenceDomain(nil)
+	registered, err := refDomain.RegisterReference(refObject)
+	assert.NoError(t, err)
+
+	resolved, err := refDomain.ResolveReference(registered)
+	assert.NoError(t, err)
+	assert.Equal(t, domain, resolved.Domain)
+	assert.Equal(t, record, resolved.Record)
+
+}
+
+func TestReferenceDomain_ResolveReference_IncorrectRef(t *testing.T) {
+	refDomain := newReferenceDomain(nil)
+	_, err := refDomain.ResolveReference("1")
+	assert.Equal(t, "object with record 1 does not exist", err.Error())
```

Comment:
You could use `assert.EqualError` for error checks.

`assert.EqualError(t, err, "object with record 1 does not exist")`

Reply:
Thx, fixed